Makefile

```diff
@@ -0,0 +1,69 @@
+# Makefile for running CI checks locally
+# Mirrors GitHub Actions workflow in .github/workflows/ci.yml
+
+.PHONY: help sync lint format types test ci verify fix clean
+
+# CI environment variables (from .github/workflows/ci.yml)
+export PIP_DISABLE_PIP_VERSION_CHECK := 1
+export COVERAGE_IGOR_VERBOSE := 1
+export FORCE_COLOR := 1
+
+# Default target
+help:
+	@echo "🔧 CI Targets (mirror GitHub Actions):"
+	@echo "  make ci       - Run ALL CI checks (lint + format + types + test)"
+	@echo "  make verify   - Sync dependencies + run all CI checks"
+	@echo "  make lint     - Run ruff linter"
+	@echo "  make format   - Check code formatting"
+	@echo "  make types    - Run pyright type checker"
+	@echo "  make test     - Run pytest with coverage"
+	@echo ""
+	@echo "🛠️  Development Targets:"
+	@echo "  make sync     - Install/sync all dependencies"
+	@echo "  make fix      - Auto-fix lint and format issues"
+	@echo "  make clean    - Remove build artifacts and caches"
+
+# Install dependencies (combines lint + test groups like CI does)
+sync:
+	uv sync --locked --group dev
+
+# === CI Sanity Checks ===
+
+lint:
+	uv run --locked ruff check --diff
+
+format:
+	uv run --locked ruff format --diff
+
+types:
+	uv run --locked pyright
+
+# === CI Tests ===
+
+test:
+	uv run --locked pytest --cov-report=xml
```

Comment:
Locally we don't want or need the XML report. The HTML report is much more useful for local development.

Reply:
Oh, that's fine then. I know it's only in CI to upload to Codecov.